keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1537 @@
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Dynamic resource allocation introduces an alternative to the existing [device
+manager
+API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-management/device-plugin.md)
+for third-party hardware vendors. Both are expected to co-exist, with vendors
+choosing the API that better suits their needs on a case-by-case basis. Because
+the new API is going to be implemented independently of the existing device
+manager, there's little risk of breaking stable APIs.
+
+The new API is inspired by the existing [volume provisioning support with CSI](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#provisioning-and-deleting) and uses similar
+concepts. The goal is to let users request resources with parameters that can
+be different depending on what kind of resource gets requested. Resource
+allocations can be ephemeral (specified in a Pod spec, allocated and destroyed
+together with the Pod) and persistent (lifecycle managed separately from a Pod,
+with an allocated resource used for multiple different Pods).
+
+Several core Kubernetes components must be modified (see the
+[implementation](#implementation) section for details):
+- kube-apiserver (new API)
+- kube-controller-manager (new controller)
+- kube-scheduler (new builtin plugin)
+- kubelet (new third-party plugin kind)
+
+Resources are managed by third-party plugins that communicate with central
+Kubernetes components, in particular the kube-scheduler, by updating
+objects stored in the kube-apiserver. kube-scheduler only needs to be modified
+once to support dynamic resource allocation. Then multiple plugins from
```

Comment:
Does this "multiple plugins" refer to third party plugin of kubelet? Better to clarify this since the scheduler has plugin too. How about renaming the "third-party plugin" to "resource plugin"?

Reply:
Now it is called "resource driver" or (when it is clear) just "driver". Such a driver has a "resource kubelet plugin" on each node and a "resource driver controller" as part of the control plane.

Reply:
OK, thanks.